### PR TITLE
(PDB-2779) README: document approach to durability

### PR DESCRIPTION
**Pull request by rbrw**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,6 +69,43 @@ root, you can invoke `./run-test --use-sudo` to do the same.  Though
...
+guaranteed to be unique, and where durability is provided by this
+sequence of operations:
+
+  - Write the entry data to a temp file,
+  - fdatasync() the temp file,
+  - rename() the temp file to the final name,
+  - and fsync() the parent directory.
+
+Or rather, stockpile calls JVM functions that are believed to provide
+that behavior or an equivialent result.  The parent directory fsync()
```
Comment (wkalt) [inline README.md]: equivalent

File: README.md [hunk 2]
```diff
@@ -69,6 +69,43 @@ root, you can invoke `./run-test --use-sudo` to do the same.  Though
...
+is required in order to ensure that the destination file name is also
+durable.
+
+Given this approach, the "fsync rate" of the underlying filesystem
+will constrain on the entry storage rate for a given queue, and so
```
Comment (wkalt) [inline README.md]: constrain the

File: README.md (hunk 1 above)
Comment (rbrw) [reply README.md]: Fixed, thanks.

File: README.md (hunk 2 above)
Comment (rbrw) [reply README.md]: Fixed, thanks.

**Comment by rbrw** [comment]
Should be merged iff we're sufficiently convinced the approach described seems sound.

**Pull request merged**